### PR TITLE
Entering edit mode for text element crash

### LINKED ISSUE #1719: Editing text broken on double click
## Bug Description

Editor crashes while entering edit mode for text element.

## Expected Behaviour

Editor should not broken

## Steps to Reproduce

1. Create text element.
2. Text edit mode, types some text including some line break.
3. Click on canvas outside of text element to exit edit mode.
4. Try entering edit mode and exit

## Screenshots

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria

Text element edit mode should work without crashes.

### PR DESCRIPTION
**Pull request by ndev1991**
## Summary

When trying edit mode of text element multiple times, crashing the editor.

## Relevant Technical Choices

EditorState of draft.js to get the state behaving in some weird way and sometimes can not get None states. Added default values for them.

## To-do

---

Editor should not crash and underline, bold, italic should not be selected when default.

Fixes #1719

### CODE REVIEW

File: assets/src/edit-story/components/richText/formatters/color.js [hunk 1]
```diff
@@ -71,7 +71,7 @@ function getColor(editorState) {
     return MULTIPLE_VALUE;
   }
   const colorStyle = styles[0];
-  if (colorStyle === NONE) {
+  if (!styles.length || colorStyle === NONE) {
```
Comment (dvoytenko) [inline assets/src/edit-story/components/richText/formatters/color.js]: Let's just do above `const colorStyle = styles[0] ?? NONE`

File: assets/src/edit-story/components/richText/formatters/italic.js [hunk 2]
```diff
@@ -43,7 +43,7 @@ function stylesToCSS(styles) {
 
 function isItalic(editorState) {
   const styles = getPrefixStylesInSelection(editorState, ITALIC);
-  return !styles.includes(NONE);
+  return Boolean(styles.length) && !styles.includes(NONE);
```
Comment (dvoytenko) [inline assets/src/edit-story/components/richText/formatters/italic.js:46]: Wouldn't we expect `styles` here to always contain `NONE` for an empty case?

/cc @barklund

File: assets/src/edit-story/components/richText/formatters/letterSpacing.js [hunk 3]
```diff
@@ -63,7 +63,7 @@ function getLetterSpacing(editorState) {
     return MULTIPLE_VALUE;
   }
   const spacingStyle = styles[0];
-  if (spacingStyle === NONE) {
+  if (!styles.length || spacingStyle === NONE) {
```
Comment (dvoytenko) [inline assets/src/edit-story/components/richText/formatters/letterSpacing.js]: Ditto: `styles[0] ?? NONE`

**Review by dvoytenko — CHANGES_REQUESTED** [review]

File: assets/src/edit-story/components/richText/formatters/color.js (hunk 1 above)
Comment (ndev1991) [reply assets/src/edit-story/components/richText/formatters/color.js]: Done

File: assets/src/edit-story/components/richText/formatters/letterSpacing.js (hunk 3 above)
Comment (ndev1991) [reply assets/src/edit-story/components/richText/formatters/letterSpacing.js]: Done

File: assets/src/edit-story/components/richText/formatters/italic.js (hunk 2 above)
Comment (barklund) [reply assets/src/edit-story/components/richText/formatters/italic.js:46]: I'm pretty sure that should happen. But there might be weird edge cases.

**Comment by barklund** [comment]
This is the wrong approach. Opening new PR with proper solution.

**Pull request closed without merging**